File: threeSum.java
import java.util.*;
public class threeSum {
    public static void main(String args[]){

    }
    public List<List<Integer>> threeSum(int[] nums) {
        int sum = 0, num1=0, num2=0, num3=0;
        
        for(int i = 0; i < nums.length; i++){
            for(int j = i; j < nums.length-1; j++){
                for(int k = j; k < nums.length-2; k++){
                    sum += nums[i] + nums[j+1] + nums[k+2];
                    if(sum == 0){
                        num1 = nums[i];
                        num2 = nums[j+1];
                        num3 = nums[k+2];
                        break;
                    }
                }
            }
        }
        
    }
}

File: threeSum.cpp
#include <iostream>
#include <vector>
#include <set>

using namespace std;
vector<vector<int>> threeSum(vector<int> &nums)
{
    int sum = 0, Vcount = 0;
    vector<vector<int>> ArrVec;
     for (int i = 0; i < nums.size(); i++)
    {
        for (int j = i; j < nums.size() - 1; j++)
        {
            for (int k = j; k < nums.size() - 2; k++)
            {
                sum += nums[i] + nums[j + 1] + nums[k + 2];
                if (sum == 0)
                {
                    ArrVec[Vcount].push_back(nums[i]);
                    ArrVec[Vcount].push_back(nums[j+1]);
                    ArrVec[Vcount].push_back(nums[k+2]);
                    Vcount++;
                }
            }
        }
    }

}
int main()
{

    return 0;
}